File: mobile/src/types/student.ts
export interface StudentSubjectRequest {
  subject_id: string;
  education_level_id: string;
}

export interface StudentProfileUpdateRequest {
  education_level_id?: string;
  bio?: string;
  subjects?: StudentSubjectRequest[];
}

export interface StudentSubjectResponse {
  id: string;
  subject_id: string;
  subject_name: string | null;
  education_level_id: string;
  education_level_name: string | null;
}

export interface StudentProfileResponse {
  account_id: string;
  education_level_id: string | null;
  education_level_name: string | null;
  bio: string | null;
  subjects: StudentSubjectResponse[];
}

File: backend/app/schemas/student.py
import uuid

from pydantic import BaseModel


class StudentSubjectRequest(BaseModel):
    subject_id: uuid.UUID
    education_level_id: uuid.UUID


class StudentProfileUpdateRequest(BaseModel):
    education_level_id: uuid.UUID | None = None
    bio: str | None = None
    subjects: list[StudentSubjectRequest] | None = None


class StudentSubjectResponse(BaseModel):
    id: uuid.UUID
    subject_id: uuid.UUID
    subject_name: str | None = None
    education_level_id: uuid.UUID
    education_level_name: str | None = None

    model_config = {"from_attributes": True}


class StudentProfileResponse(BaseModel):
    account_id: uuid.UUID
    education_level_id: uuid.UUID | None
    education_level_name: str | None = None
    bio: str | None
    subjects: list[StudentSubjectResponse] = []

    model_config = {"from_attributes": True}
